### PR TITLE
update doc for message log

### PR DESCRIPTION
**Pull request by wanlwanl**

### CODE REVIEW

File: docs/diagnostic-logs.md [hunk 1]
```diff
@@ -4,17 +4,26 @@
...
-    - [Authorization failure](#authorization-failure)
-    - [Throttling](#throttling)
+    - [Connection related issues](#connection-related-issues)
+      - [Unexpected connection number changes](#unexpected-connection-number-changes)
+        - [Unexpected connection dropping](#unexpected-connection-dropping)
+        - [Unexpected connection growing](#unexpected-connection-growing)
+      - [Authorization failure](#authorization-failure)
+      - [Throttling](#throttling)
+    - [Message related issues](#message-related-issues)
+      - [Message lost](#message-lost)
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:26]: message loss?

File: docs/diagnostic-logs.md [hunk 2]
```diff
@@ -35,34 +44,108 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
-1.	Then click **Add diagnostic setting**.
+1.	Then you will get a full view of the diagnostic settings.
 
-	![Add diagnostic logs](./images/diagnostic-logs/add-diagnostic-setting.png)
+	![Diagnostic settings' full view](./images/diagnostic-logs/azure-signalr-diagnostic-settings.png)
 
-1.	Set the archive target that you want. Currently, we support **Archive to a storage account** and **Send to Log Analytics**.
+1. Configure the log source settings.
+   1. In **Log Source Settings** section, check the the specific log type you want to collect for all connections. Otherwise the the log will be collected only for [diagnostic clients](#diagnostic-client).
+2. Configure the loc destination settings.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:53]: loc -> log

File: docs/diagnostic-logs.md [hunk 3]
```diff
@@ -35,34 +44,108 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
 
-	![Add diagnostic logs](./images/diagnostic-logs/add-diagnostic-setting.png)
+	![Diagnostic settings' full view](./images/diagnostic-logs/azure-signalr-diagnostic-settings.png)
 
-1.	Set the archive target that you want. Currently, we support **Archive to a storage account** and **Send to Log Analytics**.
+1. Configure the log source settings.
+   1. In **Log Source Settings** section, check the the specific log type you want to collect for all connections. Otherwise the the log will be collected only for [diagnostic clients](#diagnostic-client).
+2. Configure the loc destination settings. 
+   1. In **Log Destination Settings** section, a table of diagnostic settings display the existing diagnostic settings. You can click the link in the table to get access to the log destination to view the collected diagnostic logs.
+   1. In this section, click the botton **Configure Log Destination Settings** to add, update, or delete diagnostic settings.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:55]: botton -> button

File: docs/diagnostic-logs.md [hunk 4]
```diff
@@ -35,34 +44,108 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
 
-1.	Then click **Add diagnostic setting**.
+1.	Then you will get a full view of the diagnostic settings.
 
-	![Add diagnostic logs](./images/diagnostic-logs/add-diagnostic-setting.png)
+	![Diagnostic settings' full view](./images/diagnostic-logs/azure-signalr-diagnostic-settings.png)
 
-1.	Set the archive target that you want. Currently, we support **Archive to a storage account** and **Send to Log Analytics**.
+1. Configure the log source settings.
+   1. In **Log Source Settings** section, check the the specific log type you want to collect for all connections. Otherwise the the log will be collected only for [diagnostic clients](#diagnostic-client).
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:52]: Since there is only one item, no need to be a list?

File: docs/diagnostic-logs.md (hunk 2 above)
Comment (chenkennt) [inline docs/diagnostic-logs.md:53]: Since you use 1 in most of the list items, also use 1 here.

File: docs/diagnostic-logs.md [hunk 5]
```diff
@@ -35,34 +44,108 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
+#### Connectivity Logs
 
+Connectivity logs provide detailed information for SignalR hub connections. For example, basic information (user ID, connection ID and transport type, etc.) and event information (connect, disconnect and abort event, etc.). Therefore, connectivity log is helpful to troubleshoot connection related issues. For typical connection related troubleshooting guide, see [connection related issue](#connection-related-issues). 
 
-1.	Save the new diagnostics settings.
+#### Messaging Logs
 
-New settings take effect in about 10 minutes. After that, logs appear in the configured archival target, in the **Diagnostics logs** pane.
+Messaging logs provide tracing information for the SignalR hub messages received and sent via SignalR service. For example, tracing ID and message type of the message. The tracing ID and message type is also logged in app server. Typically the message is recorded when it is arrived at or left from service or server. Therefore messaging logs are helpful for troubleshooting message related issues. For typical message related troubleshooting guide, see [message related issues](#message-related-issues)
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:72]: when it ~is arrived~ arrives at or ~left~ leaves from?

File: docs/diagnostic-logs.md [hunk 6]
```diff
@@ -35,34 +44,108 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
 
-For more information about configuring diagnostics, see the [overview of Azure diagnostic logs](../azure-monitor/platform/resource-logs-overview.md).
+> This type logs are generated for every messages, if the messages are sent frequently, messaging logs might impact the performance of SignalR service. However, you can choose different collecting behaviors to minimize the performance impact. See [diagnostic logs collecting behaviors](#Diagnostic-logs-collecting-behaviors) below.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:74]: This type *of* logs?

File: docs/diagnostic-logs.md [hunk 7]
```diff
@@ -35,34 +44,108 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
 
-### Diagnostic logs categories
+### Diagnostic logs collecting behaviors
 
-Azure SignalR Service captures diagnostic logs in one category:
+There are two typical scenarios on using diagnotic logs, especially for messaging logs.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:78]: diagnotic -> diagnostic

File: docs/diagnostic-logs.md (hunk 4 above)
Comment (wanlwanl) [reply docs/diagnostic-logs.md]: I make it 2 items. If I unformat the list, the indentation will be wierd.

File: docs/diagnostic-logs.md [hunk 8]
```diff
@@ -35,34 +44,109 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
-[TODO]
+In the mean time, others may care about the performance. They are sensitive on the latency of the message, and sometimes they need to track the message in a few connections instead of all the connections for some reason.
 
-### Diagnostic logs collecting behaviors
-[TODO]
+Therefore, SignalR service provides two kinds of collecting behaviors
+* **collect all**: collect logs in all connections 
+* **collect partially**: collect logs in some specific connections
+
+> To distinguish the connections between those collect logs and those don't collect logs, SignalR service will treat some client as diagnotic client based on the diagnostic client configurations of server and client, in which the diagnostic logs always get collected, while the others don't. For more details, see [collect partially section](#collect-partially).
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:89]: diagnotic -> diagnostic

File: docs/diagnostic-logs.md [hunk 9]
```diff
@@ -35,34 +44,109 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
+
+##### Configuration guide
+
+To enable this behavior, check the checkbox in the *Types* section in the *Log Source Settings*.
+
+This behavior doesn't require you to update server side configurations. This configuration changing will always be sent to server automatically.
+
+#### Collect partially
+
+Diagnostic logs are **only** collected by [diagnostic clients](#diagnostic-client). All messages get logged inlcuding client messages in the diagnostic clients.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:105]: How about connectivity logs?

File: docs/diagnostic-logs.md [hunk 10]
```diff
@@ -35,34 +44,109 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
+
+> The limit of the diagnostic clients' number is 100.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:107]: What happens if it's over 100?

File: docs/diagnostic-logs.md [hunk 11]
```diff
@@ -35,34 +44,109 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
+// sample: mark a client as diagnostic client when it has query string "?diag=yes" in hub URL
+public IServiceProvider ConfigureServices(IServiceCollection services)
+{
+    services.AddMvc();
+    services
+        .AddSignalR()
+        .AddAzureSignalR(o =>
+        {
+            o.ConnectionString = "<YOUR_ASRS_CONNECTION_STRING>";
+            o.DiagnosticClientFilter = context => context.Request.Query["user"] == "yes";
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:135]: should be `context.Request.Query["diag"] == "yes"`

File: docs/diagnostic-logs.md [hunk 12]
```diff
@@ -176,19 +261,77 @@ Service reloading, please reconnect | Azure SignalR Service is reloading. Azure
...
+  * Path 3: Server to SignalR service 
+  * Path 4: SignalR service to client
+
+![Message path](./images/diagnostic-logs/message-path.png)
+
+For **collect all** collecting behavior:
+
+SignalR service only trace messages in direction **from server to client via SignalR service**. The tracing ID will be generated in server, the message will carry the tracing ID to SignalR service. 
+
+By checking the log in server and service side, you can easily find out whether the message is sent from server, arrives at SignalR service, and leave from SignalR service. Basically, by checking if the *received* and *sent* message are matched or not based on message tracing Id, you can tell wether the message loss issue is in server or SignalR service in this direction. For more information, see the [details](#message-flow-detail-for-path3) below.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:306]: leave -> leaves

File: docs/diagnostic-logs.md [hunk 13]
```diff
@@ -176,19 +261,77 @@ Service reloading, please reconnect | Azure SignalR Service is reloading. Azure
...
+
+> Due to the limitation of SignalR, message tracing ID can't be passed through SignalR hub. 
+
+Once the service message arrives at SignalR service, a log called `Received a <MessageType> message <MessageTracingId> from server connection <ConnectionId>.` will be generated. Then SignalR service processes the service message and deliver to the target client(s). Once the message is sent to client(s) in **Path 4**, log `Sent a message <MessageTracingId> to client connection <ConnectionId> successfully.` will be generated.
+
+In summary, the message log will be generated when message goes in and out the SignalR service and server. You can use these logs to validate whether the message is lost in these compoments or not.
+
+If a message get lost in SignalR or server, try to get the warning logs based on the message tracing ID to get the reason. If you need further help, see the [get help section](#get-help).
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:333]: What if a message is sent outside a hub method?

File: docs/diagnostic-logs.md [hunk 14]
```diff
@@ -176,19 +261,77 @@ Service reloading, please reconnect | Azure SignalR Service is reloading. Azure
...
+
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:334]: Add some concrete samples? One case I can think of is send group message before a client is added to a group. Message log will be help on this case.

File: docs/diagnostic-logs.md (hunk 13 above)
Comment (wanlwanl) [reply docs/diagnostic-logs.md:344]: added in line 304 

```
> If you want to trace message and [send messages from outside a hub](https://docs.microsoft.com/en-us/aspnet/core/signalr/hubcontext) in your app server, you need to enable **collect all** collecting behavior to collect message logs for the messages which are not originated from diagnostic clients.
```

File: docs/diagnostic-logs.md (hunk 10 above)
Comment (wanlwanl) [reply docs/diagnostic-logs.md]: Updated:
```
> The limit of the diagnostic clients' number is 100. If the number of diagnostic clients exceeds 100, the outnumbered diagnostic clients will get throttled by SignalR service.
```

File: docs/diagnostic-logs.md (hunk 14 above)
Comment (wanlwanl) [reply docs/diagnostic-logs.md:345]: Added, includes brief steps to troubleshooting.

File: docs/diagnostic-logs.md [hunk 15]
```diff
@@ -35,38 +42,113 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
-1.	Then click **Add diagnostic setting**.
+1.	Then you will get a full view of the diagnostic settings.
 
-	![Add diagnostic logs](./images/diagnostic-logs/add-diagnostic-setting.png)
+	![Diagnostic settings' full view](./images/diagnostic-logs/azure-signalr-diagnostic-settings.png)
 
-1.	Set the archive target that you want. Currently, we support **Archive to a storage account** and **Send to Log Analytics**.
+1. Configure the log source settings.
+   1. In **Log Source Settings** section, a table shows collecting behaviors for each log type. 
+   1. Check the the specific log type you want to collect for all connections. Otherwise the the log will be collected only for [diagnostic clients](#diagnostic-client).
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:51]: > Check ~the~ the specific log

File: docs/diagnostic-logs.md [hunk 16]
```diff
@@ -35,38 +42,113 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
 
-New settings take effect in about 10 minutes. After that, logs appear in the configured archival target, in the **Diagnostics logs** pane.
+#### Messaging Logs
 
-For more information about configuring diagnostics, see the [overview of Azure diagnostic logs](../azure-monitor/platform/resource-logs-overview.md).
+Messaging logs provide tracing information for the SignalR hub messages received and sent via SignalR service. For example, tracing ID and message type of the message. The tracing ID and message type is also logged in app server. Typically the message is recorded when it arrives at or leaves from service or server. Therefore messaging logs are helpful for troubleshooting message related issues. For typical message related troubleshooting guide, see [message related issues](#message-related-issues)
 
-### Diagnostic logs categories
+> This type of logs is generated for every messages, if the messages are sent frequently, messaging logs might impact the performance of SignalR service. However, you can choose different collecting behaviors to minimize the performance impact. See [diagnostic logs collecting behaviors](#Diagnostic-logs-collecting-behaviors) below.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:73]: every message ~messages~

File: docs/diagnostic-logs.md [hunk 17]
```diff
@@ -35,38 +42,113 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
+
+To enable this behavior, check the checkbox in the *Types* section in the *Log Source Settings*.
+
+This behavior doesn't require you to update server side configurations. This configuration changing will always be sent to server automatically.
+
+#### Collect partially
+
+Diagnostic logs are **only** collected by [diagnostic clients](#diagnostic-client). All messages get logged including client messages in the diagnostic clients.
+
+> The limit of the diagnostic clients' number is 100. If the number of diagnostic clients exceeds 100, the outnumbered diagnostic clients will get throttled by SignalR service.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:105]: Explain what's the behavior of connection get throttled.

File: docs/diagnostic-logs.md [hunk 18]
```diff
@@ -35,38 +42,113 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
+
+##### Configuration guide
+
+To enable this behavior, check the checkbox in the *Types* section in the *Log Source Settings*.
+
+This behavior doesn't require you to update server side configurations. This configuration changing will always be sent to server automatically.
+
+#### Collect partially
+
+Diagnostic logs are **only** collected by [diagnostic clients](#diagnostic-client). All messages get logged including client messages in the diagnostic clients.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:103]: Also mention connectivity logs

File: docs/diagnostic-logs.md [hunk 19]
```diff
@@ -35,38 +42,113 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
+   1. In this section, click the button **Configure Log Destination Settings** to add, update, or delete diagnostic settings.
+   1. Click **Add diagnostic setting** to add a new diagnostic setting, or click **Edit** to modify an existing diagnostic setting.
+   1.	Set the archive target that you want. Currently, SignalR service supports **Archive to a storage account** and **Send to Log Analytics**.
+   1. Select the logs you want to archive. Only `AllLogs` is available for diagnostic log. It only controls whether you want to archive the logs. To configure which log types needs to be generated in SignalR service, configure in **Log Source Settings** section.
+	![Diagnostics settings pane](./images/diagnostic-logs/diagnostics-settings-pane.png)
+   1.	Save the new diagnostics setting. The new setting takes effect in about 10 minutes. After that, logs will be sent to configured archival target. For more information about configuring log destination settings, see the [overview of Azure diagnostic logs](../azure-monitor/platform/resource-logs-overview.md).
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:59]: `../azure-monitor/platform/resource-logs-overview.md` should be bad url.

File: docs/diagnostic-logs.md [hunk 20]
```diff
@@ -176,28 +259,99 @@ Service reloading, please reconnect | Azure SignalR Service is reloading. Azure
...
+  * Path 3: Server to SignalR service 
+  * Path 4: SignalR service to client
+
+![Message path](./images/diagnostic-logs/message-path.png)
+
+For **collect all** collecting behavior:
+
+SignalR service only trace messages in direction **from server to client via SignalR service**. The tracing ID will be generated in server, the message will carry the tracing ID to SignalR service.
+
+> If you want to trace message and [send messages from outside a hub](https://docs.microsoft.com/en-us/aspnet/core/signalr/hubcontext) in your app server, you need to enable **collect all** collecting behavior to collect message logs for the messages which are not originated from diagnostic clients.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:304]: so there is no way to collect client to server message and message sent outside hub method at the same time?

File: docs/diagnostic-logs.md [hunk 21]
```diff
@@ -176,28 +259,99 @@ Service reloading, please reconnect | Azure SignalR Service is reloading. Azure
...
+
+> Due to the limitation of ASP.NET Core SignalR, the message comes from client doesn't contains any message level ID. But ASP.NET SignalR generate *invocation ID* for each message, you can use it to map with the tracing ID.
+
+Then the message carries the tracing ID Server in **Path 2**. Server will generate a log `Received message <messagetracingId> from client connection <connectionId>` once the message arrives.
+
+<span id="message-flow-detail-for-path3"></span>
+Once the message invokes the hub method in server, a new service message will be generated with a *new tracing ID*. Once the service message is generated, server will generate a log in template `Start to broadcast/send message <MessageTracingId> ...`, the actual log will be based on your scenario. Then the message will be delivered to SignalR service in **Path 3**, once the service message leaves from server, a log called `Succeeded to send message <MessageTracingId>` will be generated. 
+
+> Due to the limitation of SignalR, message tracing ID can't be passed through SignalR hub.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:327]: what does this mean?

File: docs/diagnostic-logs.md [hunk 22]
```diff
@@ -176,28 +259,99 @@ Service reloading, please reconnect | Azure SignalR Service is reloading. Azure
...
+In summary, the message log will be generated when message goes in and out the SignalR service and server. You can use these logs to validate whether the message is lost in these components or not.
+
+Below is a typical message loss issue.
+
+###### A user fails to receive messages in a group
+
+The typical story in this issue is that the user joins a group **after** sending a group message.
+Without diagnostic logs, you are unable to find out when the user is connected and when the group message is sent.
+Once you enable messaging logs, you are able to compare the message arriving time in SignalR service. Follow the below steps to troubleshoot:
+1. Find the message logs in server to find when the user is joined the group and when the group message is sent. 1. Get the message tracing ID A of joining the group and the message tracing ID B of group message from the message logs.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:340]: is joined -> joined

Comment (chenkennt) [reply docs/diagnostic-logs.md:340]: > 1. Get the message tracing ID A of joining the group and the message tracing ID B of group message from the message logs.
Should be in a separate line?

File: docs/diagnostic-logs.md [hunk 23]
```diff
@@ -176,28 +259,99 @@ Service reloading, please reconnect | Azure SignalR Service is reloading. Azure
...
+
+Once the service message arrives at SignalR service, a log called `Received a <MessageType> message <MessageTracingId> from server connection <ConnectionId>.` will be generated. Then SignalR service processes the service message and deliver to the target client(s). Once the message is sent to client(s) in **Path 4**, log `Sent a message <MessageTracingId> to client connection <ConnectionId> successfully.` will be generated.
+
+In summary, the message log will be generated when message goes in and out the SignalR service and server. You can use these logs to validate whether the message is lost in these components or not.
+
+Below is a typical message loss issue.
+
+###### A user fails to receive messages in a group
+
+The typical story in this issue is that the user joins a group **after** sending a group message.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:337]: Explain this problem a little bit more (for example, a sample code that customer didn't await the group join before send the message to group).

File: docs/diagnostic-logs.md (hunk 20 above)
Comment (wanlwanl) [reply docs/diagnostic-logs.md:304]: Customer can turn on collect-all and mark the client as diagnostic client at the same time. 

As we discussed offline,
1. We can make the message invoked outside of hub class has a way to get logged without turning on collect-all in the future.
2. Collect logs of client to server message as advanced option

Comment (chenkennt) [reply docs/diagnostic-logs.md:304]: > Customer can turn on collect-all and mark the client as diagnostic client at the same time.

This is not clear by reading the doc. I think customer may get confused that diagnostic client only works when he chooses to collect partially.

File: docs/diagnostic-logs.md (hunk 21 above)
Comment (wanlwanl) [reply docs/diagnostic-logs.md:327]: I want to explain the ID of message from client cannot map to the ID of service message (service message will be sent to ASRS).
How about I change to:
> The tracing ID of message from client cannot map to the tracing ID of service message to be sent to SignalR service.

File: docs/diagnostic-logs.md (hunk 20 above)
Comment (wanlwanl) [reply docs/diagnostic-logs.md:304]: I will add a new line to explain, and add a link to the diagnostic section

> Diagnostic clients works for both **collect all** and **collect partially** collecting behaviors. It has higher priority to collect logs. For more information, see [diagnostic client section](#diagnostic-client).

File: docs/diagnostic-logs.md (hunk 23 above)
Comment (wanlwanl) [reply docs/diagnostic-logs.md]: added

File: docs/diagnostic-logs.md (hunk 21 above)
Comment (chenkennt) [reply docs/diagnostic-logs.md]: This sounds better.

File: docs/diagnostic-logs.md (hunk 23 above)
Comment (wanlwanl) [reply docs/diagnostic-logs.md]: updated  for all the comments at new commit https://github.com/Azure/azure-signalr/pull/1047/commits/60b3edeff9a95bbde78b4f1a06ab19e7fd66c364 , please take a look

File: docs/diagnostic-logs.md [hunk 24]
```diff
@@ -35,38 +42,115 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
+
+Diagnostic logs are collected by all the connections. Take messaging logs for example. When this behavior is enabled, SignalR service will send a notification to server to start generating tracing ID for each message. The tracing ID will be carried in the message to the service, the service will also log the message with tracing ID.
+
+> Note that to ensure the performance of SignalR service, SignalR service doesn't await and parse the whole message sent from client, therefore, the client messages isn't get logged. But if the client is marked as a diagnostic client, then client message will get logged in SignalR service.
+
+##### Configuration guide
+
+To enable this behavior, check the checkbox in the *Types* section in the *Log Source Settings*.
+
+This behavior doesn't require you to update server side configurations. This configuration changing will always be sent to server automatically.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:99]: changing -> change

File: docs/diagnostic-logs.md [hunk 25]
```diff
@@ -35,38 +42,115 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
...
+
+> Diagnostic clients works for both **collect all** and **collect partially** collecting behaviors. It has higher priority to collect logs.
+
+##### Configuration guide
+
+To enable this behavior, you need to configure service, server, client side.
+
+###### Service side <!-- omit in toc -->
+
+To enable this behavior, uncheck the checkbox for a specific log type in the *Types* section in the *Log Source Settings*.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md:119]: This is not accurate according to the note above.

> Diagnostic clients works for both **collect all** and **collect partially** collecting behaviors. It has higher priority to collect logs.

Comment (chenkennt) [reply docs/diagnostic-logs.md:119]: Maybe move the note to here?

Comment (wanlwanl) [reply docs/diagnostic-logs.md:119]: Why move to here? The note talks about diagnostic client, put it under Diagnostic Client section is ok?

Comment (chenkennt) [reply docs/diagnostic-logs.md:119]: I mean these two paragraphs conflict with each other. Putting them together may be easier to understand.

**Review by chenkennt — APPROVED** [review]

**Pull request merged**